src/uct/ib/base/ib_iface.c

```diff
@@ -791,7 +791,6 @@ void uct_ib_iface_fill_ah_attr_from_gid_lid(uct_ib_iface_t *iface, uint16_t lid,
 
     if (iface->config.force_global_addr ||
         (iface->gid_info.gid.global.subnet_prefix != gid->global.subnet_prefix)) {
-        ucs_assert_always(gid->global.interface_id != 0);
```

Comment:
but it means we create ah_attr with global=1 and interface_id=0, doesn't it cause error during communication?
maybe we should use iface->config.force_global_addr only when packing the address, and not when creating ah_attr?

Reply:
I found that we dont pack interface id only in case of eth link layer, AFAUI `iface->config.force_global_addr` should be ignorred then since `gid.raw` is used

Reply:
in case of connect by sockaddr, rdmacm component is not aware about IB iface conffiguration and address is packed with parameters provided by rdmacm